lib/global_config.js

```diff
@@ -278,4 +278,29 @@ exports.getKeepAlive = function () {
   return keepAlive;
 };
 
+let credentialManager = createJsonFileCredentialManager();
+
+exports.setCustomCredentialManager = function (customCredentialManager) {
+  Errors.checkArgumentValid(Util.checkValidCustomCrednetialManager(customCredentialManager),
+  ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_CUSTOM_CREDENTIAL_MANAGER);
+  
+  credentialManager =  new function () {  
+    this.read = async function (key) {  
```

Comment:
why are we wrapping these functions instead of just assigning the whole object?

Reply:
I did this because users can use the async functions. However, as you mentioned, it would be better to assign the whole object and I added the 'await' command on the connection and the sf.